src/main/java/com/demo/MybatisApplication/repository/StudentRepository.java

```diff
@@ -7,17 +7,14 @@
 import org.apache.ibatis.annotations.Param;
 
 import java.util.List;
+import java.util.Map;
 
 @Mapper
 public interface StudentRepository {
 
-    void saveStudent(StudentEntity student);
-
-    void updateSubjectsToStudent(@Param("studentId") Long studentId, @Param("subjects") List<SubjectEntity> subjects);
-
+    void save(StudentEntity student);
+    void saveAllSubjectsForStudent(@Param("studentId") Long studentId, @Param("subjects") List<SubjectEntity> subjects);
     StudentEntity findStudentById(Long id);
-
-    StudentEntity findStudentWithSubjects(Long id);
-
-    List<StudentEntity> findAllStudents();
+    StudentEntity findBySubjects(Long id);
```

Comment:
Please check Manjiri and Parth's PR, I think validations are missing

Reply:
did not understand. which validations?

Reply:
I have added validations in feature/dtos PR